File: features/discovery/search/src/main/java/com/tokopedia/search/result/di/component/SearchComponent.java
package com.tokopedia.search.result.di.component;

import com.tokopedia.abstraction.common.di.component.BaseAppComponent;
import com.tokopedia.discovery.newdiscovery.di.scope.SearchScope;
import com.tokopedia.search.di.module.SearchTrackingModule;
import com.tokopedia.search.di.module.UserSessionModule;
import com.tokopedia.search.result.data.gql.initiatesearch.GqlInitiateSearchModule;
import com.tokopedia.search.result.domain.usecase.InitiateSearchUseCaseModule;
import com.tokopedia.search.result.presentation.presenter.search.SearchPresenterModule;
import com.tokopedia.search.result.presentation.view.activity.SearchActivity;

import dagger.Component;

@SearchScope
@Component(modules = {
        UserSessionModule.class,
        SearchTrackingModule.class,
        SearchPresenterModule.class
}, dependencies = BaseAppComponent.class)
public interface SearchComponent {

    void inject(SearchActivity searchActivity);
}

File: features/tokomart/src/main/java/com/tokopedia/tokomart/search/di/SearchComponent.kt
package com.tokopedia.tokomart.search.di

import com.tokopedia.abstraction.common.di.component.BaseAppComponent
import com.tokopedia.tokomart.search.presentation.view.SearchFragment
import com.tokopedia.tokomart.searchcategory.di.UserSessionModule
import dagger.Component

@SearchScope
@Component(
        modules = [
            SearchViewModelModule::class,
            SearchContextModule::class,
            SearchQueryParamModule::class,
            UserSessionModule::class,
                  ],
        dependencies = [BaseAppComponent::class])
interface SearchComponent {

    fun inject(searchFragment: SearchFragment)
}